diff --git a/node_modules/@aws-amplify/cache/lib-esm/AsyncStorageCache.d.ts b/node_modules/@aws-amplify/cache/lib-esm/AsyncStorageCache.d.ts
index e44deec..fffa631 100644
--- a/node_modules/@aws-amplify/cache/lib-esm/AsyncStorageCache.d.ts
+++ b/node_modules/@aws-amplify/cache/lib-esm/AsyncStorageCache.d.ts
@@ -1,5 +1,5 @@
 import { StorageCache } from './StorageCache';
-import { AsyncStorage } from 'react-native';
+import AsyncStorage from '@react-native-community/async-storage';
 import { ICache } from './types';
 export declare class AsyncStorageCache extends StorageCache implements ICache {
     /**
diff --git a/node_modules/@aws-amplify/cache/lib-esm/AsyncStorageCache.js b/node_modules/@aws-amplify/cache/lib-esm/AsyncStorageCache.js
index a528567..391d748 100644
--- a/node_modules/@aws-amplify/cache/lib-esm/AsyncStorageCache.js
+++ b/node_modules/@aws-amplify/cache/lib-esm/AsyncStorageCache.js
@@ -61,7 +61,7 @@ var __generator = (this && this.__generator) || function (thisArg, body) {
 };
 import { StorageCache } from './StorageCache';
 import { defaultConfig, getCurrTime } from './Utils';
-import { AsyncStorage } from 'react-native';
+import AsyncStorage from '@react-native-community/async-storage';
 import { ConsoleLogger as Logger } from '@aws-amplify/core';
 var logger = new Logger('AsyncStorageCache');
 /*
diff --git a/node_modules/@aws-amplify/cache/lib/AsyncStorageCache.d.ts b/node_modules/@aws-amplify/cache/lib/AsyncStorageCache.d.ts
index e44deec..fffa631 100644
--- a/node_modules/@aws-amplify/cache/lib/AsyncStorageCache.d.ts
+++ b/node_modules/@aws-amplify/cache/lib/AsyncStorageCache.d.ts
@@ -1,5 +1,5 @@
 import { StorageCache } from './StorageCache';
-import { AsyncStorage } from 'react-native';
+import AsyncStorage from '@react-native-community/async-storage';
 import { ICache } from './types';
 export declare class AsyncStorageCache extends StorageCache implements ICache {
     /**
diff --git a/node_modules/@aws-amplify/cache/src/AsyncStorageCache.ts b/node_modules/@aws-amplify/cache/src/AsyncStorageCache.ts
index 0b41745..08a824e 100644
--- a/node_modules/@aws-amplify/cache/src/AsyncStorageCache.ts
+++ b/node_modules/@aws-amplify/cache/src/AsyncStorageCache.ts
@@ -13,7 +13,7 @@
 
 import { StorageCache } from './StorageCache';
 import { defaultConfig, getCurrTime } from './Utils';
-import { AsyncStorage } from 'react-native';
+import AsyncStorage from '@react-native-community/async-storage';
 import { ICache } from './types';
 import { ConsoleLogger as Logger } from '@aws-amplify/core';
 
