diff --git a/node_modules/@aws-amplify/core/lib-esm/RNComponents/reactnative.d.ts b/node_modules/@aws-amplify/core/lib-esm/RNComponents/reactnative.d.ts
index 6c24337..7b80bad 100644
--- a/node_modules/@aws-amplify/core/lib-esm/RNComponents/reactnative.d.ts
+++ b/node_modules/@aws-amplify/core/lib-esm/RNComponents/reactnative.d.ts
@@ -1 +1,2 @@
-export { Linking, AppState, AsyncStorage } from 'react-native';
+export { Linking, AppState } from 'react-native';
+export AsyncStorage from '@react-native-community/async-storage';
diff --git a/node_modules/@aws-amplify/core/lib-esm/RNComponents/reactnative.js b/node_modules/@aws-amplify/core/lib-esm/RNComponents/reactnative.js
index 224ea70..182c90c 100644
--- a/node_modules/@aws-amplify/core/lib-esm/RNComponents/reactnative.js
+++ b/node_modules/@aws-amplify/core/lib-esm/RNComponents/reactnative.js
@@ -10,5 +10,6 @@
  * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
  * and limitations under the License.
  */
-export { Linking, AppState, AsyncStorage } from 'react-native';
+export { Linking, AppState } from 'react-native';
+export AsyncStorage from '@react-native-community/async-storage';
 //# sourceMappingURL=reactnative.js.map
\ No newline at end of file
diff --git a/node_modules/@aws-amplify/core/lib-esm/StorageHelper/reactnative.js b/node_modules/@aws-amplify/core/lib-esm/StorageHelper/reactnative.js
index 35080ea..cbcf408 100644
--- a/node_modules/@aws-amplify/core/lib-esm/StorageHelper/reactnative.js
+++ b/node_modules/@aws-amplify/core/lib-esm/StorageHelper/reactnative.js
@@ -10,7 +10,7 @@
  * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
  * and limitations under the License.
  */
-import { AsyncStorage } from 'react-native';
+import AsyncStorage from '@react-native-community/async-storage';
 var MEMORY_KEY_PREFIX = '@MemoryStorage:';
 var dataMemory = {};
 /** @class */
diff --git a/node_modules/@aws-amplify/core/lib/RNComponents/reactnative.d.ts b/node_modules/@aws-amplify/core/lib/RNComponents/reactnative.d.ts
index 6c24337..7b80bad 100644
--- a/node_modules/@aws-amplify/core/lib/RNComponents/reactnative.d.ts
+++ b/node_modules/@aws-amplify/core/lib/RNComponents/reactnative.d.ts
@@ -1 +1,2 @@
-export { Linking, AppState, AsyncStorage } from 'react-native';
+export { Linking, AppState } from 'react-native';
+export AsyncStorage from '@react-native-community/async-storage';
diff --git a/node_modules/@aws-amplify/core/src/RNComponents/reactnative.ts b/node_modules/@aws-amplify/core/src/RNComponents/reactnative.ts
index 9d926b5..8d43347 100644
--- a/node_modules/@aws-amplify/core/src/RNComponents/reactnative.ts
+++ b/node_modules/@aws-amplify/core/src/RNComponents/reactnative.ts
@@ -11,4 +11,5 @@
  * and limitations under the License.
  */
 
-export { Linking, AppState, AsyncStorage } from 'react-native';
+export { Linking, AppState } from 'react-native';
+export AsyncStorage from '@react-native-community/async-storage';
diff --git a/node_modules/@aws-amplify/core/src/StorageHelper/reactnative.ts b/node_modules/@aws-amplify/core/src/StorageHelper/reactnative.ts
index 37841ba..47c9db4 100644
--- a/node_modules/@aws-amplify/core/src/StorageHelper/reactnative.ts
+++ b/node_modules/@aws-amplify/core/src/StorageHelper/reactnative.ts
@@ -10,7 +10,7 @@
  * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
  * and limitations under the License.
  */
-import { AsyncStorage } from 'react-native';
+import AsyncStorage from '@react-native-community/async-storage';
 
 const MEMORY_KEY_PREFIX = '@MemoryStorage:';
 let dataMemory = {};
