diff --git a/node_modules/react-native-image-crop-picker/ios/src/ImageCropPicker.m b/node_modules/react-native-image-crop-picker/ios/src/ImageCropPicker.m
index 25ead1a..6485e53 100644
--- a/node_modules/react-native-image-crop-picker/ios/src/ImageCropPicker.m
+++ b/node_modules/react-native-image-crop-picker/ios/src/ImageCropPicker.m
@@ -453,6 +453,7 @@ - (void) handleVideo:(AVAsset*)asset withFileName:(NSString*)fileName withLocalI
                                              withRect:CGRectNull
                                      withCreationDate:nil
                                  withModificationDate:nil
+                                       withFileSource:nil
                         ]);
         } else {
             completion(nil);
@@ -479,7 +480,7 @@ - (void) getVideoAsset:(PHAsset*)forAsset completion:(void (^)(NSDictionary* ima
      }];
 }
 
-- (NSDictionary*) createAttachmentResponse:(NSString*)filePath withExif:(NSDictionary*) exif withSourceURL:(NSString*)sourceURL withLocalIdentifier:(NSString*)localIdentifier withFilename:(NSString*)filename withWidth:(NSNumber*)width withHeight:(NSNumber*)height withMime:(NSString*)mime withSize:(NSNumber*)size withDuration:(NSNumber*)duration withData:(NSString*)data withRect:(CGRect)cropRect withCreationDate:(NSDate*)creationDate withModificationDate:(NSDate*)modificationDate {
+- (NSDictionary*) createAttachmentResponse:(NSString*)filePath withExif:(NSDictionary*) exif withSourceURL:(NSString*)sourceURL withLocalIdentifier:(NSString*)localIdentifier withFilename:(NSString*)filename withWidth:(NSNumber*)width withHeight:(NSNumber*)height withMime:(NSString*)mime withSize:(NSNumber*)size withDuration:(NSNumber*)duration withData:(NSString*)data withRect:(CGRect)cropRect withCreationDate:(NSDate*)creationDate withModificationDate:(NSDate*)modificationDate withFileSource:(NSString*)fileSource {
     return @{
              @"path": (filePath && ![filePath isEqualToString:(@"")]) ? filePath : [NSNull null],
              @"sourceURL": (sourceURL) ? sourceURL : [NSNull null],
@@ -494,7 +495,8 @@ - (NSDictionary*) createAttachmentResponse:(NSString*)filePath withExif:(NSDicti
              @"cropRect": CGRectIsNull(cropRect) ? [NSNull null] : [ImageCropPicker cgRectToDictionary:cropRect],
              @"creationDate": (creationDate) ? [NSString stringWithFormat:@"%.0f", [creationDate timeIntervalSince1970]] : [NSNull null],
              @"modificationDate": (modificationDate) ? [NSString stringWithFormat:@"%.0f", [modificationDate timeIntervalSince1970]] : [NSNull null],
-             @"duration": (duration) ? duration : [NSNull null]
+             @"duration": (duration) ? duration : [NSNull null],
+             @"fileSource": (fileSource) ? fileSource : [NSNull null]
              };
 }
 
@@ -513,6 +515,8 @@ - (NSString *)determineMimeTypeFromImageData:(NSData *)data {
         case 0x49:
         case 0x4D:
             return @"image/tiff";
+        case 0x00:
+            return @"image/heic";
     }
     return @"";
 }
@@ -576,8 +580,6 @@ - (void)qb_imagePickerController:
                              @autoreleasepool {
                                  UIImage *imgT = [UIImage imageWithData:imageData];
                                  
-                                 Boolean forceJpg = [[self.options valueForKey:@"forceJpg"] boolValue];
-
                                  NSNumber *compressQuality = [self.options valueForKey:@"compressImageQuality"];
                                  Boolean isLossless = (compressQuality == nil || [compressQuality floatValue] >= 0.8);
 
@@ -589,9 +591,10 @@ - (void)qb_imagePickerController:
 
                                  NSString *mimeType = [self determineMimeTypeFromImageData:imageData];
                                  Boolean isKnownMimeType = [mimeType length] > 0;
+                                 NSLog(@"CroppingFile mimeType %@", mimeType);
 
                                  ImageResult *imageResult = [[ImageResult alloc] init];
-                                 if (isLossless && useOriginalWidth && useOriginalHeight && isKnownMimeType && !forceJpg) {
+                                 if (isLossless && useOriginalWidth && useOriginalHeight && isKnownMimeType) {
                                      // Use original, unmodified image
                                      imageResult.data = imageData;
                                      imageResult.width = @(imgT.size.width);
@@ -603,9 +606,11 @@ - (void)qb_imagePickerController:
                                  }
 
                                  NSString *filePath = @"";
+                                 NSString *fileSource = @"";
                                  if([[self.options objectForKey:@"writeTempFile"] boolValue]) {
 
                                      filePath = [self persistFile:imageResult.data];
+                                     fileSource = [self persistFile:imageResult.data];
 
                                      if (filePath == nil) {
                                          [indicatorView stopAnimating];
@@ -636,6 +641,7 @@ - (void)qb_imagePickerController:
                                                                              withRect:CGRectNull
                                                                      withCreationDate:phAsset.creationDate
                                                                  withModificationDate:phAsset.modificationDate
+                                                                       withFileSource:fileSource
                                                         ]];
                              }
                              processed++;
@@ -738,6 +744,7 @@ - (void) processSingleImagePick:(UIImage*)image withExif:(NSDictionary*) exif wi
     } else {
         ImageResult *imageResult = [self.compression compressImage:[image fixOrientation]  withOptions:self.options];
         NSString *filePath = [self persistFile:imageResult.data];
+        NSString *fileSource = [self persistFile:imageResult.data];
         if (filePath == nil) {
             [viewController dismissViewControllerAnimated:YES completion:[self waitAnimationEnd:^{
                 self.reject(ERROR_CANNOT_SAVE_IMAGE_KEY, ERROR_CANNOT_SAVE_IMAGE_MSG, nil);
@@ -762,6 +769,7 @@ - (void) processSingleImagePick:(UIImage*)image withExif:(NSDictionary*) exif wi
                                                withRect:CGRectNull
                                        withCreationDate:creationDate
                                    withModificationDate:modificationDate
+                                         withFileSource:fileSource
                           ]);
         }]];
     }
@@ -801,6 +809,7 @@ - (void)imageCropViewController:(UIViewController *)controller
     ImageResult *imageResult = [self.compression compressImage:resizedImage withOptions:self.options];
 
     NSString *filePath = [self persistFile:imageResult.data];
+    NSString *fileSource = [self persistFile:imageResult.data];
     if (filePath == nil) {
         [self dismissCropper:controller selectionDone:YES completion:[self waitAnimationEnd:^{
             self.reject(ERROR_CANNOT_SAVE_IMAGE_KEY, ERROR_CANNOT_SAVE_IMAGE_MSG, nil);
@@ -828,6 +837,7 @@ - (void)imageCropViewController:(UIViewController *)controller
                                            withRect:cropRect
                                    withCreationDate:self.croppingFile[@"creationDate"]
                                withModificationDate:self.croppingFile[@"modificationDate"]
+                                     withFileSource:fileSource
                       ]);
     }]];
 }
@@ -836,9 +846,14 @@ - (void)imageCropViewController:(UIViewController *)controller
 // we are saving image and saving it to the tmp location where we are allowed to access image later
 - (NSString*) persistFile:(NSData*)data {
     // create temp file
+    NSString *mimeType = [self determineMimeTypeFromImageData:data];
+    NSString *extension = ([mimeType isEqualToString:@"image/heic"]) ? @".heic" : @".jpg";
+
+    NSLog(@"CroppingFile mimeType 2 %@", mimeType);
+    
     NSString *tmpDirFullPath = [self getTmpDirectory];
     NSString *filePath = [tmpDirFullPath stringByAppendingString:[[NSUUID UUID] UUIDString]];
-    filePath = [filePath stringByAppendingString:@".jpg"];
+    filePath = [filePath stringByAppendingString:extension];
 
     // save cropped file
     BOOL status = [data writeToFile:filePath atomically:YES];
@@ -860,31 +875,31 @@ + (NSDictionary *)cgRectToDictionary:(CGRect)rect {
 
 #pragma mark - TOCCropViewController Implementation
 - (void)cropImage:(UIImage *)image {
-    TOCropViewController *cropVC;
-    if ([[[self options] objectForKey:@"cropperCircleOverlay"] boolValue]) {
-        cropVC = [[TOCropViewController alloc] initWithCroppingStyle:TOCropViewCroppingStyleCircular image:image];
-    } else {
-        cropVC = [[TOCropViewController alloc] initWithImage:image];
-        CGFloat widthRatio = [[self.options objectForKey:@"width"] floatValue];
-        CGFloat heightRatio = [[self.options objectForKey:@"height"] floatValue];
-        if (widthRatio > 0 && heightRatio > 0){
-            CGSize aspectRatio = CGSizeMake(widthRatio, heightRatio);
-            cropVC.customAspectRatio = aspectRatio;
+    dispatch_async(dispatch_get_main_queue(), ^{
+        TOCropViewController *cropVC;
+        if ([[[self options] objectForKey:@"cropperCircleOverlay"] boolValue]) {
+            cropVC = [[TOCropViewController alloc] initWithCroppingStyle:TOCropViewCroppingStyleCircular image:image];
+        } else {
+            cropVC = [[TOCropViewController alloc] initWithImage:image];
+            CGFloat widthRatio = [[self.options objectForKey:@"width"] floatValue];
+            CGFloat heightRatio = [[self.options objectForKey:@"height"] floatValue];
+            if (widthRatio > 0 && heightRatio > 0){
+                CGSize aspectRatio = CGSizeMake(widthRatio, heightRatio);
+                cropVC.customAspectRatio = aspectRatio;
             
+            }
+            cropVC.aspectRatioLockEnabled = ![[self.options objectForKey:@"freeStyleCropEnabled"] boolValue];
+            cropVC.resetAspectRatioEnabled = !cropVC.aspectRatioLockEnabled;
         }
-        cropVC.aspectRatioLockEnabled = ![[self.options objectForKey:@"freeStyleCropEnabled"] boolValue];
-        cropVC.resetAspectRatioEnabled = !cropVC.aspectRatioLockEnabled;
-    }
 
-    cropVC.title = [[self options] objectForKey:@"cropperToolbarTitle"];
-    cropVC.delegate = self;
+        cropVC.title = [[self options] objectForKey:@"cropperToolbarTitle"];
+        cropVC.delegate = self;
     
-    cropVC.doneButtonTitle = [self.options objectForKey:@"cropperChooseText"];
-    cropVC.cancelButtonTitle = [self.options objectForKey:@"cropperCancelText"];
+        cropVC.doneButtonTitle = [self.options objectForKey:@"cropperChooseText"];
+        cropVC.cancelButtonTitle = [self.options objectForKey:@"cropperCancelText"];
     
-    cropVC.modalPresentationStyle = UIModalPresentationFullScreen;
+        cropVC.modalPresentationStyle = UIModalPresentationFullScreen;
         
-    dispatch_async(dispatch_get_main_queue(), ^{
         [[self getRootVC] presentViewController:cropVC animated:FALSE completion:nil];
     });
 }
